### ISSUE TITLE
Git diff view persists after pressing Ctrl+C to exit

### ISSUE THREAD

**Issue by dkaksl** [git-for-windows/git#1113]
- [x] I was not able to find an [open](https://github.com/git-for-windows/git/issues?q=is%3Aopen) or [closed](https://github.com/git-for-windows/git/issues?q=is%3Aclosed) issue matching what I'm seeing

### Setup

 - Which version of Git for Windows are you using? Is it 32-bit or 64-bit?

64-bit

```
$ git --version --build-options
git version 2.9.0.windows.1
```

 - Which version of Windows are you running? Vista, 7, 8, 10? Is it 32-bit or 64-bit?

7

```
$ cmd.exe /c ver

Microsoft Windows [Version 6.1.7601]
```

 - What options did you set as part of the installation? Or did you choose the
   defaults?

Probably defaults. I'm using git-for-windows that comes bundled with [Cmder](http://cmder.net/).

 - Any other interesting things about your environment that might be related
   to the issue you're seeing?

I am specifically running the git-for-windows binary from `cmd`, so this is probably not a cmder isolated issue.

### Details

 - Which terminal/shell are you running Git from? e.g Bash/CMD/PowerShell/other

CMD

 - What commands did you run to trigger this issue? If you can provide a
   [Minimal, Complete, and Verifiable example](http://stackoverflow.com/help/mcve)
   this will help us understand the issue.

**Steps to reproduce**

1. Make some changes to any version controlled file (changes should be long enough to stay in scrollable diff view)
2. `git diff`
3. <kbd>Ctrl</kbd> + <kbd>C</kbd>
4. Press <kbd>Enter</kbd> multiple times

 - What did you expect to occur after running these commands?

Exit diff view. If that fails, to exit it when <kbd>Q</kbd> is subsequently pressed.

 - What actually happened instead?

Every second press of <kbd>Enter</kbd> brings back a diff view prompt `:` and <kbd>Q</kbd> does not allow the shell to exit.

**Comment by dscho** [comment]
This issue may be fixed in a more current version. Could you give [the current snapshot](https://wingit.blob.core.windows.net/files/index.html) a try?

**Comment by dscho** [comment]
I take that as a yes.

**Comment by dkaksl** [comment]
Just tested it now. No it is not fixed.

Every second <kbd>Ctrl</kbd> + <kbd>C</kbd> press brings back the diff view prompt `:`.

This persists even after executing `exit` to close the git shell.

**Comment by dscho** [comment]
> This persists even after executing exit to close the git shell.

Wait. What "git shell" do you close that does not close the window?

Your bug report specified very clearly that you run this in CMD, and there was no description of any "git shell" whatsoever.

**Comment by dkaksl** [comment]
@dscho the "git shell" is the interface you get from running git-cmd.exe in CMD

You then execute `exit` once to go back to the normal CMD interface (without the git commands).

**Comment by dkaksl** [comment]
@dscho is it possible for you to just try it yourself? It took me like 5 minutes from downloading the latest snapshot.

**Comment by dscho** [comment]
> the "git shell" is the interface you get from running git-cmd.exe in CMD
> 
> You then execute exit once to go back to the normal CMD interface (without the git commands).

That is not how you are supposed to use this. Either call *Git CMD* from the Start Menu, or call `git-cmd.exe` from the Windows Explorer or <kbd>Windows</kbd>+<kbd>R</kbd> prompt.

Calling CMD from within CMD is not really supported, it is hard enough to support Git CMD as-is...

**Comment by dkaksl** [comment]
Just tested on the latest snapshot, again, as you described, running git-cmd.exe from Windows Explorer. It recreates.

**Comment by dscho** [comment]
@dkaksl there must be a missing piece of information, as I really cannot reproduce the described behavior here.

**Comment by dkaksl** [comment]
OK, let me try again and attach a gif maybe...

**Comment by dscho** [comment]
For the record, I see no GIF. And I'd much rather have a detailed list of steps in accordance with http://stackoverflow.com/help/mcve and http://whathaveyoutried.com/.

**Comment by dkaksl** [comment]
I meant I would add a gif as soon as I have the opportunity.

As for my instructions on recreating the issue, I'm not sure how I could
write them more clearly.

**Comment by AraHaan** [comment]
I get this a lot on my end too making things much unreadable. I also noticed it happens more often on large doffs of like at least over 1k lines or so of changes. It also sometimes does it on small diffs though.

**Comment by dscho** [comment]
I am working on a fix for https://github.com/git-for-windows/git/issues/1491 which should also address this here issue. Once a new snapshot becomes available at https://wingit.blob.core.windows.net/files/index.html, I invite you all to test it.
